lib/vdsm/virt/backup.py

```diff
@@ -74,6 +74,9 @@ class ScratchDiskConfig(properties.Owner):
         values=[DISK_TYPE.FILE, DISK_TYPE.BLOCK])
 
     def __init__(self, **kw):
+        self.img_id = kw.get("img_id")
+        self.dom_id = kw.get("dom_id")
+        self.vol_id = kw.get("vol_id")
         self.path = kw.get("path")
         self.type = kw.get("type")
```

Comment:
Please add the keys in the same order they appear in the schema. We want to the
to have the same structure in the schema, in vdsm internal implementation, and if
possible in on engine side.

Please add properties to the class - see how path and type are added. The properties provide validation for the values. See how we define the same keys for BackupDisk.

Note that the ids are not required, since when using transient disk, we don't have
any id.

Reply:
Actually for `BackupDisk` I don't see any properties defined, unless I'm missing something :)
But I believe I understood what you suggested:
Added the 3 properties:
```
    sd_id = properties.UUID(required=False)
    img_id = properties.UUID(required=False)
    vol_id = properties.UUID(required=False)
```